pandas/_libs/algos.pyx

```diff
@@ -393,6 +393,99 @@ def nancorr_spearman(ndarray[float64_t, ndim=2] mat, Py_ssize_t minp=1) -> ndarr
     return result
 
 
+# ----------------------------------------------------------------------
+# Kendall correlation
+# Wikipedia article: https://en.wikipedia.org/wiki/Kendall_rank_correlation_coefficient
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def nancorr_kendall(ndarray[float64_t, ndim=2] mat, Py_ssize_t minp=1) -> ndarray:
```

Comment:
we usually dont mix-and-match cython-vs-python style annotations.  i know this is tough bc the python-style won't allow `ndarray[float64_t, ndim=2]`.  not sure if there's a better alternative

Reply:
Right now, the function signature for this function is the same as the one for nancorr_spearman. However, I notice that other functions have left out the `` -> ndarray`` part(including nancorr_pearson). Should I remove that from this function?

Reply:
right but we are currently not checking the .pyx right so this is moot for now?

Reply:
yah its not a big deal